src/app/api/services/initialization.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+from typing import Dict
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import yaml
+
+from app.core.settings import Settings
+
+
+def edit_yaml(original_file: Path, params: Dict[str, str], restart_file: Path) -> Path:
+    """A file to dynamically edit the YAML file to be used in T-Route
+
+    Parameters:
+    -----------
+    original_file: Path
+        The path to the base yaml config file
+    params: Dict[str, str]
+        The parameters that will be added to the base config file
+    restart_file: path
+        The location to the restart_file path
+
+    Returns:
+    --------
+    Path:
+        The path to the dynamically generated config
+    """
+    tmp_yaml = original_file.with_name(
+        original_file.stem + "_tmp_" + params["lid"] + original_file.suffix
+    )
+    with open(original_file, "r") as file:
+        data = yaml.safe_load(file)
+
+    output_dir = Path(
+        data["output_parameters"]["stream_output"]["stream_output_directory"].format(
+            params["lid"]
+        )
+    )
+    output_dir.mkdir(exist_ok=True)
+
+    data["network_topology_parameters"]["supernetwork_parameters"]["geo_file_path"] = (
+        params["geo_file_path"]
+    )
+
+    data["compute_parameters"]["restart_parameters"]["start_datetime"] = params[
+        "start_datetime"
+    ]
+    data["compute_parameters"]["restart_parameters"]["lite_channel_restart_file"] = (
+        restart_file.__str__()
+    )
+    data["compute_parameters"]["forcing_parameters"]["nts"] = params["nts"]
+    data["compute_parameters"]["forcing_parameters"]["qlat_input_folder"] = params[
+        "qlat_input_folder"
+    ]
+
+    data["output_parameters"]["stream_output"]["stream_output_directory"] = (
+        output_dir.__str__()
+    )
+
+    with open(tmp_yaml, "w") as file:
+        yaml.dump(data, file)
+
+    return tmp_yaml
+
+
+def create_params(
+    lid: str,
+    feature_id: str,
+    hy_id: str,
+    initial_start: float,
+    start_time: str,
+    num_forecast_days: int,
+    settings: Settings,
+) -> Dict[str, str]:
+    """Generating the parameters to be added to the T-Route config files
+
+    Parameters:
+    -----------
+    lid: str
+        The location ID for the RFC point
+    feature_id: str
+        The feature ID, or COMID/hf_id, from the RFC point
+    hy_id: str
+        The HYFeatures ID from the RFC point
+    initial_start: float
+        The initial starting flow (m3/s) for Routing
+    start_time: str
+        The starting time ("%Y-%m-%dT%H:%M:%S") for when routing begins
+    num_forecast_days: int
+        The number of days we are going to route
+    settings: Settings
+        The T-route BaseSettings
+
+    Returns:
+    --------
+    Dict[str, str]:
+        The parameters to be added to the t-route config file
+    """
+    dt = datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    start_datetime = dt.strftime("%Y-%m-%d_%H:%M")
+
+    nts = 288 * num_forecast_days
```

Comment:
It wouldn't be too difficult to refactor this to use forecast hours as an input, would make this more generally useful for a wider range of potential applications